client/my-sites/domains/domain-management/dataviews/dataviews-fields/domain-status-field.tsx

```diff
@@ -0,0 +1,60 @@
+import { SiteDetails } from '@automattic/data-stores';
+import { DomainsTableStatusCell } from '@automattic/domains-table/src/domains-table/domains-table-status-cell';
+import { resolveDomainStatus } from '@automattic/domains-table/src/utils/resolve-domain-status';
+import { useTranslate } from 'i18n-calypso';
+import { DomainData } from '../types';
+import { useDomainsDataViewsContext } from '../use-context';
+
+interface Props {
+	domain: DomainData;
+}
+
+const DomainStatusField = ( { domain }: Props ) => {
+	const { sites, domainResults, updatingDomain, domainStatusPurchaseActions } =
+		useDomainsDataViewsContext();
+	const translate = useTranslate();
+
+	const site =
+		sites[ domain.processed.blogId ] && ( sites[ domain.processed.blogId ] as SiteDetails );
+	const pendingUpdates = domainResults.get( domain.processed.domain ) ?? [];
+
+	if ( domain.processed.domain === updatingDomain?.domain && updatingDomain?.message ) {
+		pendingUpdates.unshift( {
+			created_at: updatingDomain?.created_at,
+			message: updatingDomain?.message,
+			status: '',
+		} );
+	}
+
+	const domainStatus = domain
+		? resolveDomainStatus( domain.processed, {
```

Comment:
Could we use item.original.domain_status.status as the domain status? It seems that we are calculating the status in the component responsible for rendering it, which isn't ideal. Ideally, this component should focus on displaying the status with appropriate styling and icons, without performing any computations.

Reply:
Unfortunately, we can't at the moment. Maybe when we switch to a custom backend.